project/Publish.scala

```diff
@@ -0,0 +1,33 @@
+package build.play.soap
+
+import sbt._
+import Keys._
+
+/**
+ * For projects that are not to be published.
+ */
+object NoPublish extends AutoPlugin {
+  override def requires = plugins.JvmPlugin
+
+  override def projectSettings = Seq(
+    skip in publish := true,
+  )
+}
+
+class Publish(releaseRepo: String, snapshotRepo: String) extends AutoPlugin {
+  import bintray.BintrayPlugin
+  import bintray.BintrayPlugin.autoImport._
+
+  override def trigger  = allRequirements
+  override def requires = BintrayPlugin
+
+  override def projectSettings = Seq(
+    bintrayOrganization := Some("playframework"),
+    bintrayRepository := (if (isSnapshot.value) snapshotRepo else releaseRepo),
+    bintrayPackage := "play-soap",
+    bintrayReleaseOnPublish := false
+  )
+}
```

Comment:
this is inspired on other projects we have, except that for play soap we have a library and a plugin to publish